microsoft/testsuites/network/netinterface.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from time import sleep
+
+from lisa import (
+    Logger,
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+    simple_requirement,
+)
+from lisa.features import NetworkInterface
+from lisa.nic import Nics
+from lisa.operating_system import CoreOs
+from lisa.tools import Dhclient, Uname, Wget
+from lisa.util import constants
+
+
+@TestSuiteMetadata(
+    area="network",
+    category="functional",
+    description="""
+    This test suite validates basic functionalities of Network Interfaces.
+    """,
+)
+class NetInterface(TestSuite):
+    RELOAD_NETVSC_TEST_COUNT = 10
+    NET_IFUP_IFDOWN_TEST_COUNT = 4
+
+    @TestCaseMetadata(
+        description="""
+            This test case verifies if synthetic network interface can be
+            brought up and brought down gracefully.
+
+            Steps:
+            1. Validate netvsc isn't built-in already. If it is then skip the test.
+            2. Unload and load netvsc module multiple times in a loop.
+            3. Change nic state to up and down multiple times.
+                Each time after "up" state, verify ip address is assigned to nic
+                and internet is accessible via nic.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            min_nic_count=1,
+            network_interface=schema.NetworkInterfaceOptionSettings(
+                data_path=schema.NetworkDataPath.Synthetic,
+            ),
+        ),
+    )
+    def validate_network_interface_ifup_ifdown(self, node: Node, log: Logger) -> None:
+        self._validate_netvsc_built_in(node)
+        network_interface_feature = node.features[NetworkInterface]
+
+        # Test loading and unloading netvsc driver
+        test_count = 0
+        while test_count < self.RELOAD_NETVSC_TEST_COUNT:
+            test_count += 1
+            # Unload hv_netvsc
+            network_interface_feature.reload_netvsc()
+
+        # Clean all dhclient processes, get IP & try ping
+        node_nic_info = Nics(node)
+        node_nic_info.initialize()
+        default_nic = node_nic_info.default_nic
+        default_route = node_nic_info.default_nic_route
+
+        node_nic_info.change_nic_state(default_nic, "up")
+        if not node_nic_info.nics[default_nic].ip_addr:
+            node.execute("kill $(pidof dhclient)", shell=True, sudo=True)
+            dhclient = node.tools[Dhclient]
+            dhclient.release()
+            dhclient.renew()
+            sleep(15)
+
+        self._ping_check
+
+        test_count = 0
+        while test_count < self.NET_IFUP_IFDOWN_TEST_COUNT:
+            self.__log.info(f"NET IFUP IFDOWN Test Iteration : {test_count}")
+            node_nic_info.change_nic_state(default_nic, "down")
```

Comment:
Can it be brought back, if it's down? Does it need any extra validation after down?

Reply:
if you take down eth0 then ssh will be disconnected, you may consider to bring down and up in one command.